src/MassTransit/Courier/Contexts/TimeoutExecuteContext.cs

```diff
@@ -0,0 +1,130 @@
+namespace MassTransit.Courier.Contexts
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+
+
+    public class TimeoutExecuteContext<TArguments> :
```

Comment:
Okay. I see what you mean here. I think it would be good to create an additional constructor that accepts a CancellationToken, and then either use the inbound context.CancellationToken, or the provided one. This would make the ExecuteContextProxy() more flexible (same with Compensate) and eliminate the duplication.

Reply:
Never mind, I remember now.